typescript/agentkit/src/wallet-providers/privyWalletProvider.ts

```diff
@@ -0,0 +1,162 @@
+/**
+ * Configuration options for the Privy wallet provider.
+ *
+ * @interface
+ */
+interface PrivyWalletConfig {
+  /** The Privy application ID */
+  appId: string;
+  /** The Privy application secret */
+  appSecret: string;
+  /** The ID of the wallet to use, if not provided a new wallet will be created */
+  walletId?: string;
+  /** Optional chain ID to connect to */
+  chainId?: string;
+  /** Optional authorization key for the wallet API */
+  authorizationPrivateKey?: string;
+  /** Optional authorization key ID for creating new wallets */
+  authorizationKeyId?: string;
+}
+
+type PrivyWalletExport = {
+  walletId: string;
+  authorizationPrivateKey: string | undefined;
+  chainId: string | undefined;
+};
+
+/**
+ * A wallet provider that uses Privy's server wallet API.
+ * This provider extends the ViemWalletProvider to provide Privy-specific wallet functionality
+ * while maintaining compatibility with the base wallet provider interface.
+ */
+export class PrivyWalletProvider extends ViemWalletProvider {
+  #walletId: string;
+  #authorizationPrivateKey: string | undefined;
+
+  /**
+   * Private constructor to enforce use of factory method.
+   *
+   * @param walletClient - The Viem wallet client instance
+   * @param config - The configuration options for the Privy wallet
+   */
+  private constructor(
+    walletClient: WalletClient,
+    config: PrivyWalletConfig & { walletId: string }, // Require walletId in constructor
+  ) {
+    super(walletClient);
+    this.#walletId = config.walletId; // Now guaranteed to exist
+    this.#authorizationPrivateKey = config.authorizationPrivateKey;
+  }
+
+  /**
+   * Creates and configures a new PrivyWalletProvider instance.
+   *
+   * @param config - The configuration options for the Privy wallet
+   * @returns A configured PrivyWalletProvider instance
+   *
+   * @example
+   * ```typescript
+   * const provider = await PrivyWalletProvider.configureWithWallet({
+   *   appId: "your-app-id",
+   *   appSecret: "your-app-secret",
+   *   walletId: "wallet-id",
+   *   chainId: "84532"
+   * });
+   * ```
+   */
+  public static async configureWithWallet(config: PrivyWalletConfig): Promise<PrivyWalletProvider> {
+    const privy = new PrivyClient(config.appId, config.appSecret, {
+      walletApi: config.authorizationPrivateKey
+        ? {
+            authorizationPrivateKey: config.authorizationPrivateKey,
+          }
+        : undefined,
+    });
+
+    let walletId: string;
+    let address: `0x${string}`;
+
+    if (!config.walletId) {
+      if (config.authorizationPrivateKey && !config.authorizationKeyId) {
+        throw new Error(
+          "authorizationKeyId is required when creating a new wallet with an authorization key, this can be found in your Privy Dashboard",
+        );
+      }
+
+      try {
+        const wallet = await privy.walletApi.create({
+          chainType: "ethereum",
+          authorizationKeyIds: [config.authorizationKeyId],
```

Comment:
@azf20 `config.authorizationKeyId` could be `null` here, we need to optionally pass it:
```
authorizationKeyIds: config.authorizationKeyId ? [config.authorizationKeyId] : undefined,
```

Reply:
good catch, done!